### PR TITLE
Use go 1.23.3

### PR DESCRIPTION
**Pull request by kke**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,8 +1,8 @@
 module github.com/k0sproject/bootloose
 
-go 1.22.0
+go 1.23.0
 
-toolchain go1.22.4
+toolchain go1.23.3
```
Comment (twz123) [inline go.mod:5]:
```suggestion
go 1.23
```

Can we omit the toolchain statement?

File: go.mod [hunk 2]
```diff
@@ -1,8 +1,8 @@
 module github.com/k0sproject/bootloose
 
-go 1.22.0
+go 1.23.0
```
Comment (twz123) [inline go.mod:3]:
```suggestion
go 1.23
```

Comment (kke) [reply go.mod]: The version is now not pinned so binaries will be magically built with whatever 1.23 the CI has.

Maybe the version should be pinned in the workflow?

**Review by twz123 — APPROVED** [review]
Yeah, I don't like that "let me auto-install the toolchain for you" thing and the "pinning to patch versions" in the mod file. I think the minimum supported Go version is good, but the *actual* toolchain should be at the caller's discretion :shrug: But what shall I say, Go decided differently and I have to remember to use GOTOOLCHAIN=local everywhere :laughing:

**Pull request merged**